test/mass_act_jump_tests.jl

```diff
@@ -80,6 +80,7 @@ u0           = [0,0,0,0]
 u0[something(findfirst(isequal(:DNA),rs.syms),0)] = 1
 expected_avg = 5.926553750000000e+02
 rates = [.5, (20*log(2.)/120.), (log(2.)/120.), (log(2.)/600.), .025, 1.]
+gen_jumps!(rs)
```

Comment:
I think it's going to be needed. Doing eval in the same function as the generated function is used will give world age issues, so the user needs two separate calls.

Reply:
I'll wait a bit for @TorkelE or @korsbo to chime in with their thoughts on changing the interface. If no one objects or brings up other issues I'll then update the tests and get this ready for merging.

One other possibility would be to leave the `reaction_network` macro as constructing everything, but then provide a second macro that can be used in the query-based version. The `Union` based `make_type` code could still be used for both, so this basically just amounts to making a second macro function and corresponding simplified `coordinate` function.

Reply:
(Been elsewhere all day but reading through things now. Will have a proper reply within a few hours. Biggest thing is trying to understand the world age problem and why that approach won't work out.)

Reply:
No rush, better to get this right than hurry it! We don't want to have to change the interface again in the near future...

Reply:
In case it helps since it is no longer showing on the test, the world age error arises when using the `DiffEqJump` `Direct` method with `ConstantRateJumps` when the `fill_cur_rates` function is called. This function recursively splats out and evaluates one jump rate at a time until they've all been evaluated. I believe recursion was used instead of looping through the rate functions for performance and type stability reasons. However, this seems to then induce the world age issue since the jump rate functions are newer than the world in which `fill_cur_rates` ends up living.

The bigger issue is that even if we can fix this specific case, I guess there is the potential that (future) code within `ODEProblem`, `SDEProblem` or `JumpProblem` could do something similar, leading to the same error.